lib/vmpooler/api/v1.rb

```diff
@@ -1063,6 +1074,45 @@ def invalid_pool(payload)
       JSON.pretty_generate(result)
     end
 
+    post "#{api_prefix}/poolreset/?" do
+      content_type :json
+      result = { 'ok' => false }
+
+      if config['experimental_features']
+        need_token! if Vmpooler::API.settings.config[:auth]
+
+        begin
+          payload = JSON.parse(request.body.read)
+        rescue JSON::ParserError
+          status 400
+          result = {
+            'ok' => false,
+            'message' => 'JSON payload could not be parsed'
+          }
+        end
```

Comment:
Since the `begin`/`rescue` block ends here, won't the status always ends up getting overwritten with `404`, since `payload` will be `nil`? Perhaps the `begin` block should also include the `payload` handling?

Reply:
Good point! Another change incoming.